psi4/driver/procrouting/proc.py

```diff
@@ -988,6 +989,46 @@ def select_mp4(name, **kwargs):
         return func(name, **kwargs)
 
 
+def select_adc2(name, **kwargs):
+    """Function selecting the algorithm for ADC(2) excited state energy
+    call and directing to specified or best-performance default modules.
+
+    """
+    reference = core.get_option('SCF', 'REFERENCE')
+    mtd_type = core.get_global_option('MP_TYPE')  # TODO or ADC_TYPE or CC_TYPE or ...
+    module = core.get_global_option('QC_MODULE')
+    # Considering only adcc/adc
+
+    # TODO Actually one should do selection on a couple of other options here
+    #      as well, e.g. adcc supports frozen-core and frozen-virtual,
+    #      spin-specific states or spin-flip methods.
+
+    func = None
+    if reference == 'RHF':
+        if mtd_type == 'CONV':
+            if module == 'ADCC' and extras.addons("adcc"):
+                func = run_adcc
+            elif module in ['', 'BUILTIN']:
+                func = run_adc
+        # TODO Does the builtin support any other MP type?
```

Comment:
may switch `BUILTIN` to `ADC`, but leave for now.

No, I'm pretty sure it's adc2 only.

Reply:
I'm not sure if switching `BULTIN` to `ADC` is such a good idea, because then you have `module` being either `ADC` or `ADCC`, which is pretty easy to mix up in my opinion.

Ah, sorry, what I meant was whether the builtin supports anything else, but `CONV`, but that has been answered before ... it only supports `CONV`.